iib/workers/tasks/build.py

```diff
@@ -700,6 +706,30 @@ def _prepare_request_for_build(
         from_index_resolved = None
         ocp_version = 'v4.5'
 
+    if source_from_index:
+        with set_registry_token(overwrite_from_index_token, source_from_index):
+            source_from_index_resolved = _get_resolved_image(source_from_index)
+            source_ocp_version = (
+                get_image_label(source_from_index_resolved, 'com.redhat.index.delivery.version')
+                or 'v4.5'
+            )
+    else:
+        source_from_index_resolved = None
+        source_ocp_version = 'v4.5'
+
+    if target_from_index:
+        with set_registry_token(overwrite_from_index_token, target_from_index):
+            target_from_index_resolved = _get_resolved_image(target_from_index)
```

Comment:
This code seems to be duplicated in the same function. How about creating a helper method?

Reply:
Hmm it's only two lines that are duplicated everywhere. And they are both variable value setters. Does it really make sense to create a helper function?